wcomponents-core/src/main/java/com/github/bordertech/wcomponents/validation/AbstractWFieldIndicator.java

```diff
@@ -52,19 +52,26 @@ public AbstractWFieldIndicator(final WComponent relatedField) {
 	 * @param severity A Diagnostic severity code. e.g. {@link Diagnostic#ERROR}
 	 */
 	protected void showIndicatorsForComponent(final List<Diagnostic> diags, final int severity) {
-		FieldIndicatorModel model = getOrCreateComponentModel();
-		model.diagnostics.clear();
-		UIContext uic = UIContextHolder.getCurrent();
+		FieldIndicatorModel model = getComponentModel();
+		if (model != null) {
+			model.diagnostics.clear();
```

Comment:
This `model.diagnostic.clear()` could potentially update the `default` model as it has been retrieved with `getComponentModel` instead of `getOrCreateComponentModel`. As per your other changes, determine if the list has changed first and then get the model and update it.